src/main/java/com/denizenscript/depenizen/bukkit/objects/superiorskyblock/SuperiorSkyblockIslandTag.java

```diff
@@ -0,0 +1,397 @@
+        // @returns ElementTag(Decimal)
+        // @plugin Depenizen, SuperiorSkyblock
+        // @description
+        // Returns the current average rating of the specified island.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "total_rating", (attribute, object) -> {
+            return new ElementTag(object.getIsland().getTotalRating());
+        });
+
+        // <--[mechanism]
+        // @object SuperiorSkyblockIslandTag
+        // @name balance
+        // @input ElementTag(Decimal)
+        // @plugin Depenizen, SuperiorSkyblock
+        // @description
+        // Changes an island's bank balance.
+        // @tags
+        // <SuperiorSkyblockIslandTag.balance>
+        // -->
+        tagProcessor.registerMechanism("balance", false, ElementTag.class, (object, mechanism, value) -> {
+            if (mechanism.requireDouble()) {
+                object.getIsland().getIslandBank().setBalance(value.asBigDecimal());
+            }
+        });
+
+        // <--[mechanism]
+        // @object SuperiorSkyblockIslandTag
+        // @name description
+        // @input ElementTag
+        // @plugin Depenizen, SuperiorSkyblock
+        // @description
+        // Changes an island's description. Leave blank to remove.
+        // @tags
+        // <SuperiorSkyblockIslandTag.description>
+        // -->
+        tagProcessor.registerMechanism("description", false, ElementTag.class, (object, mechanism, value) -> {
+            object.getIsland().setDescription(value.asString());
+        });
+
+        // <--[mechanism]
+        // @object SuperiorSkyblockIslandTag
+        // @name locked
+        // @input ElementTag(Boolean)
+        // @plugin Depenizen, SuperiorSkyblock
+        // @description
+        // Changes whether an island is locked to visitors.
+        // @tags
+        // <SuperiorSkyblockIslandTag.locked>
+        // -->
+        tagProcessor.registerMechanism("locked", false, ElementTag.class, (object, mechanism, value) -> {
+            if (mechanism.requireBoolean()) {
+                object.getIsland().setLocked(value.asBoolean());
+            }
+        });
+
+        // <--[mechanism]
+        // @object SuperiorSkyblockIslandTag
+        // @name name
+        // @input ElementTag
+        // @plugin Depenizen, SuperiorSkyblock
+        // @description
+        // Changes an island's name. Multiple islands cannot have the same name.
+        // @tags
+        // <SuperiorSkyblockIslandTag.name>
+        // -->
+        tagProcessor.registerMechanism("name", false, ElementTag.class, (object, mechanism, value) -> {
+            if (mechanism.getValue().asString().isEmpty()) {
+                mechanism.echoError("You cannot have an island with no name.");
+            }
+            else if (SuperiorSkyblockAPI.getIsland(value.asString()) != null) {
+                mechanism.echoError("There is already an island with the name '" + value.asString() + "'.");
```

Comment:
Nitpick, but no need for `#asString` here (Java automatically calls `#toString` on objects when concatenating them with strings)